Add Junit 5 and new DSL syntax

checketts opened the pull request:
This is a work in progress proposal for some alternative DSLs

Commit:
Add Junit 5 and new DSL syntax

checketts commented on src/test/kotlin/cz/petrbalat/spring/mvc/test/dsl/controller/DslControllerTestV2.kt:
```diff
@@ -0,0 +1,156 @@
+            }
+        }
+
+
```
The example from lines 53-66 is my preferred new syntax.  The HttpMethod and url are required arguments. Then `builder` and `actions` expose the raw MockHttpServletRequestBuilder and ResultActions. Then we also add in a targeted `expect` and matching builder so we can clarify the expectations better (like the simplified jsonPath example on line `42`).

petrbalat replied:
It looks better than it was before. I would take a second option mockMvc.request(PUT, "/hello").
The first option mockMvc.request { PUT("/hello") {... is only for generality?

checketts replied:
I'll delete the `mockMvc.request { PUT("/hello") {` option. It was an experiment. But I don't like it because:

- adds an extra level of braces
- isn't obvious how to use, since the compiler doesn't prompt the user for the missing parts

Commit:
Fix builder, expects calls to add to the lists correctly

Commit:
Break out the separate tests

petrbalat commented on src/test/kotlin/cz/petrbalat/spring/mvc/test/dsl/controller/DslControllerTestV2.kt:
```diff
@@ -102,28 +102,28 @@ class MockMvcDslBuilder(internal var url: String? = null,
 @RequestDsl
 class DslRequestBuilder(internal var url: String? = null,
```
maybe is **url** useless?

checketts replied:
Removed. It was from an experiment with letting the URL be declared in the block, but since it is a parameter on the `request` call now. I'll remove it.

petrbalat reviewed (commented):
Its looks nice. I would like prefer only one option to create request. The second with 2 parameters.

Thx

Commit:
Move new dsl into its own file

Commit:
Make gradlew executable

Commit:
Ignore output directory

Commit:
Unify the DSL tests

checketts commented:
OK I converted a number of my project's tests to try the new DSL and added in the rest of the `expects` calls. So it feels complete now.

If it looks good. I'll update the readme with examples.

checketts commented:
In using the new DSL. Everything autocompleted very nicely with the exception of the `HttpMethod` argument. (IntelliJ was auto suggesting other packages if I tried to do shortcuts) 

While I could add `requestGet` and similarly named methods, how would you feel about converting the existing `performGet` and similar to use the new DSL blocks?  It would be a breaking API change, but I think it would make the DSL simpler and better overall.

petrbalat commented:
I don't have a problem with it. Maybe the method MockMvc.request would be better with name performRequest or just perform? Because it is similar to original method name perform in MockMvc.
